### DISCUSSION TITLE
Detect if user has manually reloaded a page?

### DISCUSSION THREAD

**Discussion by kNoAPP** [jonataslaw/getx#2046] [Q&A]
I am setting up password resets with an authCode. For example:
https://myapp.com/#/auth/authAction?code=validcode

This page current works when opened from outside my app or from another page inside my app. The needed controllers initialize, check the auth code, then display either a valid or invalid page.

However, if a user is already on the `/auth/authAction` page and they change the `code` parameter, the GetxControllers don't reinitialize and the code is never rechecked. The user stays on the current valid/invalid page. But the parameter `code` does update as I see Getx putting a debug log in that looks like this:
```
[GETX] GOING TO ROUTE /auth/action?code=validcode
[GETX] Instance "AuthActionController" has been created
[GETX] Instance "AuthActionResetPasswordController" has been created
[GETX] Instance "AuthActionResetPasswordController" has been initialized
[GETX] Instance "AuthActionController" has been initialized
** Now I manually change the page url in Chrome and hit enter **
[GETX] Redirect to null
[GETX] GOING TO ROUTE /auth/action?code=invalidcode
```

How do I listen for page reloads? I'd like to...
- Deconstruct and reconstruct my controllers on a page reload.
- If that's not possible, listen from inside the controller of a page reload and recheck the auth code.
- If that's not possible, perhaps a middleware solution?

**Comment by kNoAPP** [comment]
Still stuck on this.

**Comment by wildsurfer** [comment]
Also looking for the proper solution

**Comment by inyong1** [comment]
Replace the pageview with StateleWidget and use GetBuilder to init the controller, also dont forget to use tag that always uniq

Like this (for example)
![image](https://github.com/jonataslaw/getx/assets/5109308/17f68fc5-441f-4918-b5cc-7ae8ff7fb74e)

**Comment by inyong1** [comment]
Second solution using getView and middleware
1. create a method in controller to initlizing data
2. create middleware and call the controller method in pageBuilt
![image](https://github.com/jonataslaw/getx/assets/5109308/c637c2e7-c428-44e1-9d94-d1d2b4b2f22f)
